### ISSUE TITLE
prototype testing

### ISSUE THREAD

**Issue by mrentsch65** [mk28/TINF21C_Team2_AAS_digital_nameplate#61]
Hello, I tried to test your prototypical frontend here, but I couldn't get a response from the servers:

https://mk28.github.io/TINF21C_Team2_AAS_digital_nameplate/#/list?server=https%3A%2F%2Fadmin-shell-io.com%2F5001&search=Wittenstein_aas_xNNkj1e

https://mk28.github.io/TINF21C_Team2_AAS_digital_nameplate/#/list?server=aas.murrelektronik.com

**Comment by mk28** [comment]
Dear Customer, 
it seems that the servers you wanted to access are not AAS-Servers accessible via a REST-API. 
The adminshell-io server has a different address for their REST demo than their UI explorer. 

Please use the address `https://admin-shell-io.com/51411` for further testing.
[https://mk28.github.io/TINF21C_Team2_AAS_digital_nameplate/#/list?server=https://admin-shell-io.com/51411](https://mk28.github.io/TINF21C_Team2_AAS_digital_nameplate/#/list?server=https://admin-shell-io.com/51411)

I was not able to contact a REST-API on `aas.murrelektronik.de` as requests made seem to result in an error on their side.

![grafik](https://user-images.githubusercontent.com/32399022/201490737-b0b79a09-0200-4029-8d1f-94872fc91b9c.png)

**Comment by mrentsch65** [comment]
This is the Basyx-AAS-API:
https://app.swaggerhub.com/apis/BaSyx/basyx_asset_administration_shell_http_rest_api/v1
This is the IDTA-AAS-API:
https://app.swaggerhub.com/apis/Plattform_i40/Entire-API-Collection/V1.0RC02#/

I believe the problem is that you insert "server" in the endpoint path and use a nonstandard endpoint  "listaas":

For example, this works: 
https://aasrepo.digitaltwin-vdma.org/repo/shells
https://admin-shell-io.com/5001/shells
http://aas.murrelektronik.de:5001/shells

but not when called from your client like that: 
https://aasrepo.digitaltwin-vdma.org/repo/server/listaas
https://admin-shell-io.com/5001/server/listaas
http://aas.murrelektronik.de:5001/server/listaas

**Comment by mk28** [comment]
Dear customer @mrentsch65 ,

all three mentioned server URLs do not expose a full API according to AAS-Spec found at https://v3.admin-shell-io.com/swagger/index.html. The only endpoint exposed seems to be the registry endpoint and the shells endpoint. 

We use the `/submodels` endpoint that is described in the AAS-Spec. This does not seem to be exposed on any routes. We also use the `/shells` endpoint that is exposed in the third link. If the `/submodels` endpoint was available at `http://aas.murrelektronik.com:4001/aas/submodels` you would have to use `http://aas.murrelektronik.com:4001/aas/` to access the server. 

Another issue is that your server does not support http**s**. Browsers block requests from a secure (https) context to an insecure (http) one. This is happening here, as GitHub Pages does not allow us to host an insecure site. The request is blocked by the browser. 

Both issues together prevent any content from being shown. We cannot take any action in this case as both issues are out of our control and a problem with the server.

**Comment by mrentsch65** [comment]
Dear @mk28,
the problem is that many of these v1-servers are around and need to be served. And I do not  quite understand your problem, because both the `/shells` and the `/submodels` endpoint is obviously supported, although the url paths might be different to V3: 

http://aas.murrelektronik.com:4001/aas/shells

http://aas.murrelektronik.com:4001/aas/shells/https%3A%2F%2Faas.murrelektronik.com%2FV000-PS024-0100001%2Faas%2F1%2F0/aas/submodels

![image](https://user-images.githubusercontent.com/50681355/232194700-92819d5a-57d8-4a2b-8c00-3bf4dc5039fe.png)

What prevents you from probing for a V1 response via i.e. the `/registry` endpoint and use the alternative paths accordingly?

https is not an issue here, because these read-only AAS-Servers are open accessible anyway.

**Comment by mrentsch65** [comment]
Manipulate via "edge://flags/#unsafely-treat-insecure-origin-as-secure"  this flag:

![image](https://user-images.githubusercontent.com/50681355/232198424-f1184471-2928-4c24-8a79-dba8e091ff8e.png)

The browser will then deliver responses from the unsecure server, but this query "?level=Deep" is not supported:

![image](https://user-images.githubusercontent.com/50681355/232198160-6bcdd327-5b45-4423-ac68-a57e980175ff.png)

**Comment by mk28** [comment]
Dear Customer, 
we use the pure `/submodels` endpoint. Your API only exposes the `/shells` and all related endpoints.
The `submodels` you see in your browser is actually `/aas/shells/{assetId}/aas/submodels` which the browser shortens to `/submodels`.

Our current implementation does not use the registry because we found it to be unrelieable. https://admin-shell-io.com/5001/registry/shell-descriptors should give a list of shell descriptors that we could use similar to https://v3.admin-shell-io.com/registry/shell-descriptors but for some reason for the V1 API it is just empty and not usable, at least on the IDTA servers. 

Please provide a URL that can be used in the Murrelektronik AAS UI to open a Server as I currently have no luck opening a server with this UI. It does not produce any error messages and just does nothing if I enter URLs that should be working.
![grafik](https://user-images.githubusercontent.com/32399022/232200761-2b9488b5-6be8-4452-95ac-37c0dc629c3d.png)

With this we can investigate further.

**Comment by mrentsch65** [comment]
Returned content of the IDTA's V3-Server is not displayed:

![image](https://user-images.githubusercontent.com/50681355/232272048-6f5097cf-e896-4d27-931a-7d754a8f6a86.png)

**Comment by mrentsch65** [comment]
With the newest changes for Api-V1-support, some regressions have been introduced, i.e. selection of some servers is not longer working: 

![image](https://user-images.githubusercontent.com/50681355/232324767-f3c5eeb0-06dd-4a4c-bd31-04d36efc9244.png)

After returning to home screen or switching to server, the yellow message line should disappear or change content.

Server-URLs are not properly checked for correctness:
![image](https://user-images.githubusercontent.com/50681355/232324935-f5bc34c6-1702-494d-a424-2ba60f1d0907.png)

**Comment by mk28** [comment]
Dear customer @mrentsch65 ,
regarding your last few issues:

- The `?level=deep` is currently reqired beacause the `/submodels` would otherwise just retrun a shortened version of the submodel data. The V1 API retruns the full data by default. As it is a browser query, it is ignored, when a server cannot use the option. It should do no harm having it there for backwards compatibility. 
- I could reproduce the bug with the warning label persisiting onto the home screen and created an Issue for it.  #106 
- I could not reproduce the error that some servers could not be loaded. I reqire more Information and /  or detailed steps to reproduce it. The network tab is not always the best way to see, if the app is working correctly as some requests are rather slow.
- It is correct that there is no complex logic for validating server addresses in place. An issue for this task is created #107
- We still need the Information requested in an earlier comment about the Murrelektronik AAS server and GUI
>Please provide a URL that can be used in the Murrelektronik AAS UI to open a Server as I currently have no luck opening a server with this UI. It does not produce any error messages and just does nothing if I enter URLs that should be working.
![grafik](https://user-images.githubusercontent.com/32399022/232200761-2b9488b5-6be8-4452-95ac-37c0dc629c3d.png)
>With this we can investigate further.

As this server does not support the full swagger API spec, we are forced to mostly rewrite our server connection code once again in order to support this server. This will take time as previous issues stated that we could use the full API spec documented at  https://v3.admin-shell-io.com/swagger/index.html.
